pageserver/src/lib.rs

```diff
@@ -49,8 +50,8 @@ pub struct PageServerConf {
 /// is separate from PostgreSQL timelines, and doesn't have those
 /// limitations. A zenith timeline is identified by a 128-bit ID, which
 /// is usually printed out as a hex string.
-#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
-pub struct ZTimelineId([u8; 16]);
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct ZTimelineId(#[serde(with = "hex")] [u8; 16]);
```

Comment:
This is going to conflict with my PR: this struct needs to be serialized as binary for the wal_service networking, so I think we should add to/from hex as a separate function.